app/eth2wrap/valcache.go

```diff
@@ -132,3 +136,57 @@ func (c *ValidatorCache) Get(ctx context.Context) (ActiveValidators, CompleteVal
 
 	return resp, eth2Resp.Data, nil
 }
+
+// GetBySlot fetches active and complete validator by slot populating the cache.
+func (c *ValidatorCache) GetBySlot(ctx context.Context, slot uint64) (ActiveValidators, CompleteValidators, error) {
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	opts := &eth2api.ValidatorsOpts{
+		State:   strconv.Itoa(int(slot)),
+		PubKeys: c.pubkeys,
+	}
+
+	var eth2Resp *eth2api.Response[map[eth2p0.ValidatorIndex]*eth2v1.Validator]
+	var err error
+	maxRetries := 20
```

Comment:
I suggest we always put any constants like this one and the one below to file-level constants.